### ISSUE TITLE
RStudio does not honor the %APPDATA% environment variable, making it non-portable

### ISSUE THREAD

**Issue by mlell** [rstudio/rstudio#6903]
I am trying to set up an installation of RStudio for our group that is portable and does not alter the system it's being run on. By this approach, we hope to encapsulate all files needed for an analysis into one folder, which can be moved to another Windows computer and archived. Moreover, this way, multiple analyses running different versions of R and RStudio can coexist, improving reproducibility. 

Currently, I see no way to change the folder where RStudio is storing its preferences, other than in a way that breaks other things (see below).

# Reproducible example

The bug can be tested using RStudio 1.2.5042 by performing the following steps:

* Install RStudio, for example to `C:\RStudio`. 
* Make a new, "portable home directory" `C:\RStudio\home`
* Create the Batch script `C:\RStudio\portable.bat` to demonstrate the approach:
  ```
  SET APPDATA=%~dp0\home\AppData
  SET CSIDL_APPDATA=%APPDATA%
  start bin/rstudio.exe
  ```
  `CSIDL_APPDATA` was an attempt to fix this using information from #698, but it also does not work
 * Start RStudio by double-clicking on `portable.bat` and change a global setting (for example, pane layout or R version)

# Expected behaviour

Folders and files are created in `C:\RStudio\home\AppData` that store the changed setting. Other RStudio installations are not affected

# Actual behaviour

Files in `%USERPROFILE%\AppData` are changed which changes the behaviour of other RStudio installations. The only folders generated is an empty folder tree to `C:\RStudio\home\AppData\R\rsconnect\accounts`

# Proposal

RStudio should honor the environment variable `APPDATA` completely for all files that are under `AppData\Roaming` and also have an option to customize the location of `AppData\Local`. A list of affected folders is in the (non-working) workaround below.

# Alternative approach, works but creates other problems

Change `%USERPROFILE%` as well. Write `portable.bat` like this:

```
SET APPDATA=%~dp0\home\AppData
SET CSIDL_APPDATA=%APPDATA%
SET USERPROFILE=%~dp0\home
START bin/rstudio.exe
```

# Actual behaviour
The following directories are created in `C:\RStudio\home\AppData`:
```
├───Local
│   ├───Microsoft
│   │   ├───Internet Explorer
│   │   └───Windows
│   │       ├───Explorer
│   │       ├───History
│   │       └───INetCache
│   └───RStudio-Desktop
│       ├───client-state
│       ├───ctx
│       ├───dictionaries
│       │   ├───custom
│       │   └───languages-user
│       ├───explorer-cache
│       ├───log
│       ├───monitored
│       │   ├───connection_history
│       │   ├───lists
│       │   ├───snippets
│       │   └───user-settings
│       ├───notebooks
│       ├───pcs
│       ├───presentation
│       ├───profiles-cache
│       ├───projects
│       │   └───none
│       │       └───console06
│       ├───projects_settings
│       ├───rversion-settings
│       ├───sessions
│       │   └───active
│       ├───sources
│       │   └───per
│       │       ├───t
│       │       └───u
│       └───viewer-cache
├───R
│   └───rsconnect
│       └───accounts
└───Roaming
    └───RStudio
```

Also, settings do not affect other RStudio installations. However, this causes errors when using any file dialog (I translate from german): 

```
"C:\RStudio\home\Desktop" refers to a location that is unavailable..."
```

which make sense because I just changed the user profile folder which is where the desktop normally is.

**Comment by jmcphers** [comment]
RStudio 1.4 (currently in alpha) implements the XDG standard, so this is now possible if you use the XDG standard environment variables (it may also respect APPDATA; recommend some testing). You can try builds here:

https://dailies.rstudio.com/

**Comment by mlell** [comment]
I repeated the tests with RStudio 1.4.366 from the daily build website.

I started RStudio with the following batch script
```
SET XDG_CONFIG_HOME=%~dp0home\AppData\Roaming
rem SET APPDATA=%~dp0home\AppData\Roaming
rem SET LOCALAPPDATA=%~dp0home\AppData\Roaming
START RStudio\bin\rstudio.exe
```
The resulting file tree in the desired folder

```
AppData
└───Roaming
    └───RStudio
        └───dictionaries
            └───custom
```
So there are still some files that cannot be relocated using `XDG_CONFIG_HOME`:

* most notably the file `%APPDATA%\RStudio\desktop.ini` that contains information about the used R version.
* the file tree under `%APPDATA%\R`  (honors `APPDATA`, see below)

I removed commented out sections of the batch script above to test the effect of other environment variables. Setting `APPDATA` relocated the files under `AppData\Roaming\R`. Setting `LOCALAPPDATA` had no effect on the files in `AppData\Local\RStudio`.

So the paths to the different config files still seem to be constructed following different logic...
